nodeup/pkg/model/network.go

```diff
@@ -70,15 +70,15 @@ func (b *NetworkBuilder) Build(c *fi.ModelBuilderContext) error {
 	if networking.Cilium != nil {
 		// systemd v238 includes the bpffs mount by default; and gives an error "has a bad unit file setting" if we try to mount it again (see mount_point_is_api)
 		var alreadyMounted bool
-		_, err := os.Stat("/sys/fs/bpf")
+		// bpffs magic number
+		magic := uint32(0xCAFE4A11)
+		var fsdata unix.Statfs_t
+		err := unix.Statfs("/sys/fs/bpf", &fsdata)
+
 		if err != nil {
-			if os.IsNotExist(err) {
-				alreadyMounted = false
-			} else {
-				return fmt.Errorf("error checking for /sys/fs/bpf: %v", err)
-			}
+			alreadyMounted = false
 		} else {
-			alreadyMounted = true
+			alreadyMounted = int32(magic) == int32(fsdata.Type)
```

Comment:
What happens if the fs type is not BPF? Would it be better to return with an error also?

The Cilium code describes the logic for mounting the BPF fs and there is also a mention of a fallback location `/run/cilium/bpffs`:
https://github.com/cilium/cilium/blob/cbd63be46569751ac45060e54f6787f1286b75cd/pkg/bpf/bpffs_linux.go#L213-L224

Reply:
If BPF is not mounted on `/sys/fs/bpf`, the path will still exist, which is why the old check fails. So most likly the fs type will not be bpf before we install the unit.

`/run/cilium/bpffs` is a fallback if the host is not mounting bpf. This will cause network disruptions every time cilium pods restart though since the bpf mounts are lost.

Reply:
OK, more clear now. Thanks for clarifying the fallback part.

Regarding the check, if the path exists, `unix.Statfs` would succeed and will return `SYSFS_MAGIC=0x62656572`  or something similar. If there is an error other than the "no such file or directory", maybe it's not a good thing to continue. Same goes for a magic number that is not SYSFS_MAGIC or BPF_FS_MAGIC.

Reply:
Shouldn't you be able to mount bpffs on whatever path that exists?

I will add back the return on error shortly. `unix.Statfs("/sys/fs/bpf", &fsdata)` would return error on file not found too, shouldn't it? So the os.Stat check would be redundant.

Reply:
Just checked and you are right, if the mount path doesn't exist, the unit won't help. Probably for any type of error BPF mounting will fail.

Reply:
This part is mostly ok. There may be a rare case where the magic number is different from `SYSFS_MAGIC=0x62656572` or `BPF_FS_MAGIC= 0xCAFE4A11` and mounting will fail. I don't think it's worth adding it for now.